os/board/rtl8730e/src/rtl8730e_st7701.c

```diff
@@ -0,0 +1,250 @@
+FAR struct mipi_dsi_device *mipi_dsi_device_register(FAR struct mipi_dsi_host *host, FAR const char *name, int channel);
+FAR struct lcd_dev_s *st7701_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct st7701_config_s *config);
+
+struct st7701_config_s g_rtl8730_st7701_config_s = {
+	.init = rtl8730_st7701_lcd_init,
+	.reset = rtl8730_st7701_gpio_reset,
+	.cache_invalidate = rtl8730_st7701_cache_invalidate,
+	.lcd_reload = rtl8730_st7701_lcd_reload,
+	.lcd_enable = rtl8730_st7701_enable_lcdc,
+	.lcd_layer_enable = rtl8730_st7701_lcd_layer_enable,
+	.lcd_put_area = rtl8730_st7701_lcd_put_area,
+};
+
+struct irq {
+	u32 num;
+	u32 data;
+	u32 priority;
+};
+
+LCDC_TypeDef *pLCDC = LCDC;
+LCDC_InitTypeDef lcdc_init_struct;
+static u32 UnderFlowCnt = 0;
+
+struct irq lcdc_irq_info = {
+	.num = LCDC_IRQ,
+	.data = (u32) LCDC,
+	.priority = INT_PRI_MIDDLE,
+};
+
+extern struct irq mipi_irq_info;
+u8 *lcd_img_buffer1 = (u8 *)(DDR_BASE + LCDC_IMG_BUF_OFFSET2 + LCDC_IMG_BUF_SIZE);
+
+static void LcdcInitValues(struct lcd_data config)
+{
+	LCDC_StructInit(&lcdc_init_struct);
+	lcdc_init_struct.LCDC_ImageWidth = config.XPixels;
+	lcdc_init_struct.LCDC_ImageHeight = config.YPixels;
+	for (u8 idx = 0; idx < LCDC_LAYER_MAX_NUM; idx++) {
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgFormat = LCDC_LAYER_IMG_FORMAT_RGB565;
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgBaseAddr = (u32) lcd_img_buffer1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStop = config.XPixels;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStop = config.YPixels;
+	}
+}
+
+static void rtl8730_st7701_lcd_init(void)
+{
+	rtl8730_st7701_lcd_layer_enable(LCD_LAYER, true);
+	rtl8730_st7701_lcd_layer_enable(1, false);
+	rtl8730_st7701_lcd_layer_enable(2, false);
+
+	LCDC_Init(pLCDC, &lcdc_init_struct);
+	LCDC_DMAModeConfig(pLCDC, LCDC_LAYER_BURSTSIZE_1X64BYTES);
+	LCDC_DMADebugConfig(pLCDC, LCDC_DMA_OUT_DISABLE, 0);
+
+	rtl8730e_st7701_register_lcdc_isr();
+	
+	LCDC_LineINTPosConfig(pLCDC, YRES * 4 / 5);
+	LCDC_INTConfig(pLCDC, LCDC_BIT_LCD_LIN_INTEN | LCDC_BIT_DMA_UN_INTEN, ENABLE);
+}
+
+static void rtl8730_st7701_gpio_reset(void)
+{
+	return;
+}
+
+static void rtl8730_st7701_cache_invalidate(u32 *buffer, int size)
+{
+	DCache_CleanInvalidate((u32) buffer, size);
+}
+
+static void rtl8730_st7701_lcd_reload(void)
+{
+	LCDC_TrigerSHWReload(pLCDC);
+}
+
+static void rtl8730_st7701_lcd_layer_enable(int layer, bool enable)
+{
+	if (enable) {
+		lcdc_init_struct.layerx[layer].LCDC_LayerEn = ENABLE;
+	} else {
+		lcdc_init_struct.layerx[layer].LCDC_LayerEn = DISABLE;
+	}
+}
+
+static void rtl8730_st7701_lcd_put_area(u8 *lcd_img_buffer, u32 x_start, u32 y_start, u32 x_end, u32 y_end)
+{
+	lcdc_init_struct.layerx[LCD_LAYER].LCDC_LayerImgBaseAddr = (u32) lcd_img_buffer;
+#ifdef ENABLE_ST7701_CHANGE_WINDOW_SIZE
```

Comment:
```
#ifndef ENABLE_ST7701_CHANGE_WINDOW_SIZE
	lcdc_init_struct.layerx[LCD_LAYER].LCDC_LayerHorizontalStart = x_start;
	lcdc_init_struct.layerx[LCD_LAYER].LCDC_LayerHorizontalStop = x_end;
	lcdc_init_struct.layerx[LCD_LAYER].LCDC_LayerVerticalStart = y_start;
	lcdc_init_struct.layerx[LCD_LAYER].LCDC_LayerVerticalStop = y_end;
#else
	(void) x_start;
	(void) y_start;
	(void) x_end;
	(void) y_end;
#endif
```

Reply:
May I know where will you define ENABLE_ST7701_CHANGE_WINDOW_SIZE?

Reply:
I will change the naming to CONFIG_ENABLE_ST7701_CHANGE_WINDOW_SIZE

Reply:
I have changed this.